src/core/server/http/logging/get_response_log.ts

```diff
@@ -45,7 +45,12 @@ export function getEcsResponseLog(request: Request, log: Logger): LogMeta {
 
   // eslint-disable-next-line @typescript-eslint/naming-convention
   const status_code = isBoom(response) ? response.output.statusCode : response.statusCode;
-  const responseHeaders = isBoom(response) ? response.output.headers : response.headers;
+
+  // shallow clone the headers so they are not mutated if filtered by a RewriteAppender
+  const requestHeaders = { ...request.headers };
+  const responseHeaders = isBoom(response)
+    ? { ...response.output.headers }
+    : { ...response.headers };
```

Comment:
NIT: is shallow clone sufficient here? for array headers, the appender could potentially still mutates the values. Not sure if this is a real risk in practice though?

Reply:
Yeah that's a good point, @restrry and I were debating the tradeoffs of perf vs preventing mutation as this is a pretty hot code path.

It seems odd that someone would want to overwrite only one specific array item as they'd need to do it by index - the appender config would need to be `{ path: 'http.request.headers[0]', value: 'whatever' }`. But of course it is possible so I suppose it doesn't hurt to spread out the array here.

Reply:
Updated to prevent mutating array headers, just to be safe